backend/app/infrastructure/database/user_antisybil.py

```diff
@@ -0,0 +1,31 @@
+from typing import Optional
+from datetime import datetime
+import json
+
+from app.infrastructure.database.models import GPStamps
+from app.infrastructure.database.user import get_by_address, add_user
+from app.exceptions import UserNotFound
+from app.extensions import db
+
+
+def add_score(user_address: str, score: str, expires_at: datetime, stamps) -> GPStamps:
+    user = get_by_address(user_address)
+    if user is None:
+        add_user(user_address)
+        db.session.commit()
```

Comment:
why adding commit here? Doesn't it mean that when it fails after we will save incomplete state?

Reply:
Autogenerated ids are generated on commit. They are not available before that. If we were using `address` as primary field instead of relying on id, this would not be necessary.

Reply:
ok, but is it possible that a user does not exist in a db when we tried to save user's GP score? I don't think so, as we save the score only for a primary address (which must have signed ToS before this operation)

Reply:
Yeah, I'm changing that.